cranelift-reader/src/parser.rs

```diff
@@ -614,15 +613,29 @@ impl<'a> Parser<'a> {
     }
 
     // Match and consume either a hexadecimal Uimm128 immediate (e.g. 0x000102...) or its literal list form (e.g. [0 1 2...])
-    fn match_uimm128_or_literals(&mut self, controlling_type: Type) -> ParseResult<V128Imm> {
-        if self.optional(Token::LBracket) {
+    fn match_constant_data(&mut self, controlling_type: Type) -> ParseResult<ConstantData> {
+        let expected_size = usize::try_from(controlling_type.bytes())
+            .map_err(|_| self.error("unable to convert controlling type size to usize"))?;
```

Comment:
Maybe it's safe to just do a raw cast with `controlling_type.bytes() as usize` here?

Reply:
I would bet that it is but I remember @sunfishcode wanting me to use `try_from` for a size conversion like this so I just went with that; should I take it out?

Reply:
Considering the number of bytes is at most 128 right now (so it can be stored in a u8), and we're converting into a usize (so at least 32 bits, with a max value of `2**32 - 1`), I think we're pretty safe for a while here :)
(It could become an issue if we added a way to deserialize arbitrary user-defined structures in the future, since they could have unbounded size, but that's probably not going to happen any time soon.)